app/components/badges/view.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module Badges
+  class View < GovukComponent::Base
+    FULL_WIDTH_STATES = %w[
+      draft
+      submitted_for_trn
+      trn_received
+    ].freeze
+
+    NARROW_STATES = %w[
+      recommended_for_award
+      awarded
+      eyts_recommended
+      eyts_received
+      qts_recommended
+      qts_received
+      deferred
+      withdrawn
+    ].freeze
+
+    def initialize(trainees)
+      @trainees = trainees
```

Comment:
Does this component need to know about the database models? Would be good to keep them as simple as possible, would also make fixture generation easier for the previews/tests if you could pass in the data they need. If this is intended as a high order component then i think we need to start putting them in a specific directory to separate them from their simple counterparts